internal/k8sdeps/kunstructuredfactory.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sdeps
+
+import (
+	"io"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/kustomize/pkg/ifc"
+)
+
+// KustKunstructuredFactory hides construction using apimachinery types.
+type KustKunstructuredFactory struct {
+	decoder ifc.Decoder
+}
+
+var _ ifc.KunstructuredFactory = &KustKunstructuredFactory{}
+
+// NewKustKunstructuredFactory returns a factory.
```

Comment:
What is the meaning of `KustKunstructuredFactory`? Do you mean `KunstructuredFactory`?

Reply:
renamed to KunstructuredFactoryImpl